packages/office-ui-fabric-react/src/components/Button/_base/Button.base.tsx

```diff
@@ -0,0 +1,265 @@
+
+  constructor(props: IButtonBaseProps, rootClassName: string) {
+    super(props);
+
+    this._labelId = getId();
+    this._descriptionId = getId();
+    this._ariaDescriptionId = getId();
+  }
+
+  public render(): JSX.Element {
+    const {
+      theme,
+      className,
+      disabled,
+      href,
+      checked,
+      primary,
+      getStyles,
+      expanded,
+      onRenderIcon = this._onRenderIcon,
+      onRenderAriaDescription = this._onRenderAriaDescription,
+      onRenderChildren = this._onRenderChildren,
+      onRenderMenuIcon = this._onRenderMenuIcon,
+      onRenderPrefix,
+      onRenderSuffix
+    } = this.props;
+
+    // Anchor tag cannot be disabled hence in disabled state rendering
+    // anchor button as normal button
+    const renderAsAnchor: boolean = !disabled && !!href;
+
+    const nativeProps = this._getNativeProps(this.props, renderAsAnchor);
+
+    const {
+      ariaDescribedBy,
+      ariaLabelledBy,
+      ariaLabel
+    } = this._getAriaProps(
+        this.props,
+        getNativeProps(this.props, [renderAsAnchor ? 'a' : 'button']
+        ));
+
+    this._classNames = getClassNames(getStyles!, { theme: theme!, className, disabled, checked, expanded, primary });
+
+    let RootType = renderAsAnchor ? 'a' : 'button';
+    return (
+      <div className={ this._classNames.root }>
+        { onRenderPrefix && onRenderPrefix() }
+
+        <RootType
+          {...nativeProps}
+          className={ this._classNames.button }
+          type={ !renderAsAnchor && 'button' }
+          disabled={ disabled }
+          data-is-focusable={ ((this.props as any)['data-is-focusable'] === false || disabled) ? false : true }
+          aria-pressed={ checked }
+          aria-label={ ariaLabel }
+          aria-labelledby={ ariaLabelledBy }
+          aria-describedby={ ariaDescribedBy }
+        >
+          { onRenderIcon(this.props, this._onRenderIcon) }
+          { this._onRenderTextContents(this.props) }
+          { onRenderAriaDescription(this.props, this._onRenderAriaDescription) }
+          { onRenderChildren(this.props, this._onRenderChildren) }
+          { onRenderMenuIcon(this.props, this._onRenderMenuIcon) }
+        </RootType>
+
+        { onRenderSuffix && onRenderSuffix() }
+      </div>
+    );
+  }
+
+  public focus(): void {
+    if (this._buttonElement) {
+      this._buttonElement.focus();
+    }
+  }
+
+  public dismissMenu(): void {
+    this.setState({ menuProps: null });
+  }
+
+  @autobind
+  private _getNativeProps(props: IButtonBaseProps, renderAsAnchor: boolean) {
+    return getNativeProps(
+      props,
+      renderAsAnchor ? anchorProperties : buttonProperties,
+      ['disabled']
+    );
+  }
+
+  @autobind
+  private _onRenderIcon(props: IButtonBaseProps): JSX.Element | null {
+    return props.iconProps ? <Icon className={ this._classNames.icon } {...props.iconProps} /> : null;
+  }
+
+  @autobind
+  private _onRenderMenuIcon(props: IButtonBaseProps): JSX.Element | null {
```

Comment:
Is there a way we can move this to just be in the menu button component?

Reply:
the idea here is that you might not be using the menu button, but still want that other icon. I could change the name to prefix icon and suffix icon, or something like that. Lots of cases where icon just needs to be on the right.